proto/metadata.proto

```diff
@@ -88,9 +88,8 @@ message RangeTreeNode {
   // Color is black if true, red if false.
   optional bool black = 2 [(gogoproto.nullable) = false];
 
-  // TODO(bram): this comment has rotted, parent key is not nullable
   // If the parent key is null, this is the root node.
-  optional bytes parent_key = 3 [(gogoproto.casttype) = "Key"];
+  optional bytes parent_key = 3 [(gogoproto.customtype) = "Key"];
```

Comment:
when you move these away from customtype, please also remove the following methods from `Key` and `EncodedKey`:
```go
func (k Key) Marshal() ([]byte, error)
func (k *Key) Unmarshal(bytes []byte) error
func (k *Key) UnmarshalJSON(bytes []byte) error
func (k Key) Size() int
func (k *Key) MarshalTo(data []byte) (int, error)
```